portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java

```diff
@@ -103,19 +105,19 @@ private void pollForUnassignedJob() throws IOException {
      * Updates a unassigned {@link PortabilityJob} in storage with the provided {@code jobId} in
      * CREDS_ENCRYPTION_KEY_GENERATED state with {@code publicKey} and {@code privateKey}.
      */
-    private void keyGenerated(UUID jobId, PublicKey publicKey, PrivateKey privateKey)
+    private void keyGenerated(UUID jobId, PublicKey publicKey)
             throws IOException {
         // Lookup the job so we can append to its existing properties.
         PortabilityJob existingJob = store.findJob(jobId);
         // Verify no worker key
         // Populate job with keys to persist
-        String encodedPublicKey = PublicPrivateKeyPairGenerator.encodeKey(publicKey);
-        String encodedPrivateKey = PublicPrivateKeyPairGenerator.encodeKey(privateKey);
+        Preconditions.checkArgument(existingJob.jobAuthorization().encryptedPublicKey() == null);
+        // Populate job with public key to persist
+        String encodedPublicKey = BaseEncoding.base64Url().encode(publicKey.getEncoded());
 
         PortabilityJob updatedJob = existingJob.toBuilder()
                 .setAndValidateJobAuthorization(existingJob.jobAuthorization().toBuilder()
                         .setEncryptedPublicKey(encodedPublicKey)
```

Comment:
not related to this PR but while you're in here, can you change this to 'setEncodedPublicKey'?

actually, does it even still need to be encoded? was that just so we could pass to the client as a backup, and are we still doing that?

Reply:
Done. At some point, we will